### PR TITLE
test(jQuery): run tests with jQuery 2.1, 2.2 & 3.2

### PR DESCRIPTION
**Pull request by mgol**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Test update.

**What is the current behavior? (You can also link to an open issue here)**
We test with jQuery 2.1, 2.2 & 3.1.

**What is the new behavior (if this is a feature change)?**
We test with jQuery 2.1, 2.2 & 3.2.

**Does this PR introduce a breaking change?**
No.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] ~~Tests for the changes have been added (for bug fixes / features)~~
- [x] ~~Docs have been added / updated (for bug fixes / features)~~

**Other information**:
Also, update AngularJS 1.5 mentions in the docs to 1.6.

### CODE REVIEW

File: docs/content/tutorial/step_09.ngdoc [hunk 1]
```diff
@@ -38,16 +38,16 @@ Since we are using [Bower][bower] to install client-side dependencies, this step
   "license": "MIT",
   "private": true,
   "dependencies": {
-    "angular": "1.5.x",
-    "angular-mocks": "1.5.x",
-    "angular-route": "1.5.x",
+    "angular": "1.6.x",
+    "angular-mocks": "1.6.x",
+    "angular-route": "1.6.x",
```
Comment (gkalpak) [inline docs/content/tutorial/step_09.ngdoc]: Upgrading the tutorial form 1.5.x to 1.6.x requires more changes. There are probably more, but off the top of my head:

- The code on https://github.com/angular/angular-phonecat needs to be updated to match the changes here.
- Some `href`s (on some steps) need to take into account the new default hashPrefix (`!`).
- The code that explicitly sets the hash prefix to `!` needs to be removed (and maybe some text referring to it).

In any case, I would not make the tutorial changes as part of this PR.

**Review by gkalpak — COMMENTED** [review]
LGTM except for the tutorial related bits.

File: docs/content/tutorial/step_09.ngdoc (hunk 1 above)
Comment (mgol) [inline docs/content/tutorial/step_09.ngdoc]: OK, I'll revert them here.

Do you think I can perhaps leave jQuery 3.1 -> 3.2 update? That shouldn't require any changes.

Comment (gkalpak) [inline docs/content/tutorial/step_09.ngdoc]: In the tutorial, it was `2.2.x` --> `3.2.x`. Is 1.5.x compatible with jQuery 3.x?

Comment (mgol) [inline docs/content/tutorial/step_09.ngdoc]: Yes, AngularJS 1.5 is tested with jQuery 2.1.4, 2.2.4 & 3.1.0.

Comment (mgol) [inline docs/content/tutorial/step_09.ngdoc]: But jQuery follows semver since 1.12/2.2 so there are no breaking changes in 3.2.1.

**Comment by mgol** [comment]
This PR is now blocked on #15851.

**Comment by mgol** [comment]
The PR is now unblocked, PTAL.

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**